### ISSUE TITLE
Reponse from express

### ISSUE THREAD

**Issue by vinniimiranda** [lukeautry/tsoa#435]
I need access to the reponse from express inside a controller, i've tried the @Reponse decorator but did not work, if i can't have this i will have to stop using tsoa.

**Comment by dgreene1** [comment]
Can you describe what kind of apiroute you’re trying to write?

This feature has never been asked for because if you return an object from your tsoa route function, then the result will be set to the response body.

So everything you want to do with the Express response should be possible.

If however you want to have an explicit handle on the response then you can use Koa which allows you to get req and res.

Long story short, if you can (a) show some of your code and (b) describe your goal, we can help you to write that in a tsoa-like manner.

**Comment by vinniimiranda** [comment]
I need res to send back a report generated by jsreport, here's an example:
![image](https://user-images.githubusercontent.com/33462844/63511637-1d882900-c4b8-11e9-9963-d83c1f92d6b2.png)

Here's an example working with access of the res from 'express' 


![image](https://user-images.githubusercontent.com/33462844/63512188-90de6a80-c4b9-11e9-8c73-bda212ccf52f.png)

**Comment by dgreene1** [comment]
Here is an example of someone doing that same thing with tsoa: https://github.com/lukeautry/tsoa/issues/44#issuecomment-509693245

**Comment by vinniimiranda** [comment]
Thanks for the quickly reply but this solution did not work for me, thanks anyway.

**Comment by sundowndev** [comment]
This is still an issue. Any update?

**Comment by tsimbalar** [comment]
This is probably a bit old ... 
Maybe that's not officially supported, but if your controller returns a `Readable` (i.e. a Stream) or a `Buffer`, then it "just works". 

see also #38 #40 where support for `Buffer`s was introduced
and #551 for the support of `Readable`

**Comment by artur-klesun-st** [comment]
For me `req.res` approach from https://github.com/lukeautry/tsoa/issues/44#issuecomment-509693245 worked perfectly in Express:
```ts
@Tags('Other')
@Hidden()
@Security('api_key', ['test'])
@Get('/test')
public async test(@Request() request: express.Request): Promise<void> {
  request.res.setHeader('content-type', 'text/plain');
  request.res.write('hello');
  request.res.end();
}
```
